fvm/src/kernel/mod.rs

```diff
@@ -21,38 +21,68 @@ use fvm_shared::sys::out::vm::MessageContext;
 use fvm_shared::sys::SendFlags;
 use fvm_shared::{ActorID, MethodNum};
 
+mod blocks;
 mod hash;
 
-mod blocks;
 pub mod default;
+pub mod filecoin;
 
 pub(crate) mod error;
 
+use ambassador::delegatable_trait;
 pub use error::{ClassifyResult, Context, ExecutionError, Result, SyscallError};
 use fvm_shared::event::StampedEvent;
 pub use hash::SupportedHashes;
 use multihash::MultihashGeneric;
+use wasmtime::Linker;
 
 use crate::call_manager::CallManager;
 use crate::gas::{Gas, GasTimer, PriceList};
 use crate::machine::limiter::MemoryLimiter;
 use crate::machine::Machine;
+use crate::syscalls::InvocationData;
 
 pub struct CallResult {
     pub block_id: BlockId,
     pub block_stat: BlockStat,
     pub exit_code: ExitCode,
 }
 
+pub trait ConstructKernel<C> {
```

Comment:
So... can we:

1. Leave these on `Kernel`.
2. Not implement the delegate logic on the top-level kernel.

Or does that not make sense?

Reply:
makes sense!